src/components/ShareResultsModal/EmailField.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import * as z from 'zod';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import CheckMarkCircleSVG from './CheckMarkCircleSVG';
+
+const emailSchema = z.object({
+  email: z.string().email({ message: 'Please enter a valid email address' }),
+});
+
+export default function EmailField({ onSubmit }: Props) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful },
+  } = useForm<{ email: string }>({
+    resolver: zodResolver(emailSchema),
+  });
+
+  return (
+    <div className='cio-share-results-feature-group'>
+      <form onSubmit={handleSubmit(({ email }) => onSubmit(email))}>
+        <div className='cio-share-results-description'>Share by email</div>
+        <div className='cio-share-results-button-group'>
+          <div className='cio-share-results-email-input-group'>
+            <input
+              className={`cio-share-results-email-input ${
+                errors.email ? 'cio-share-results-email-input--error' : ''
+              }`}
+              {...register('email')}
+            />
+            {errors.email && (
+              <div className='cio-share-results-email-input-error-message'>
+                {errors.email.message}
+              </div>
+            )}
+          </div>
+          <button className='cio-share-results-share-button' type='submit'>
+            Send
+          </button>
+        </div>
+      </form>
+      {isSubmitSuccessful && (
```

Comment:
If `onEmailResults` callback throws an error, we don't display anything on the page. Is this expected? Do we want the enduser to handle that case? (I'm not against it, just checking)

Reply:
That's a good question, there are no designs for the error state unfortunately 🤔 but I'll check with Mariano in case he has some thoughts on this.

Reply:
![CleanShot 2023-10-31 at 11 19 29@2x](https://github.com/Constructor-io/constructorio-ui-quizzes/assets/7874219/58dc8cff-2fba-4654-832d-4f4cca3a130d)

Added an error state here 👍

Reply:
85a550b81a9df26541cac20a5d7d5af28d154ad3